python-threatexchange/threatexchange/signal_type/signal_base.py

```diff
@@ -150,6 +150,39 @@ def match_file(self, path: pathlib.Path) -> t.List[SignalMatch]:
         return self.match(path.read_text())
 
 
+class StrHasher(HashMatcher, StrMatcher):
+    """
+    This class can turn hashes into intermediary representations
+    """
+
+    @classmethod
+    def hash_from_str(cls, content: str) -> str:
+        """Get a string representation of the hash from a string"""
+        raise NotImplementedError
+
+    def match(self, content: str) -> t.List[SignalMatch]:
+        str_hash = self.hash_from_str(content)
+        return self.match_hash(str_hash)
+
+
+class FileHasher(HashMatcher, FileMatcher):
+    """
+    This class can hash files.
+
+    If also inheiriting from StrHasher, put this second in the inheiretence
+    to prefer file hashing to reading the file in as a Str.
+    """
+
+    @classmethod
+    def hash_from_file(self, file: pathlib.Path) -> str:
+        """Get a string representation of the hash from a file"""
+        raise NotImplementedError
+
+    def match_file(self, path: pathlib.Path) -> t.List[SignalMatch]:
```

Comment:
Not following this comment.

This class can be added to a signal type to show that it can match on arbitrary representations (hashes), and that it can convert files into arbitrary hashes. If both those are true, there is no reason to have an independent implementation for matching.

Now is probably a good time to say I plan on removing all the match functions from SignalType, so this problem will probably solve itself.